### PR TITLE
fix(jsr): ensure proper storage of ahead of time module infos

### PR DESCRIPTION
**Pull request by dsherret**
This was incorrectly being stored so the AOT cache for JSR specifiers wasn't being used. I added a wrapper type to help prevent the API being used incorrectly.

### CODE REVIEW

File: cli/cache/mod.rs [hunk 1]
```diff
@@ -280,10 +282,11 @@ impl Loader for FetchCacher {
     source: &str,
     module_info: &deno_graph::ModuleInfo,
   ) {
```
Comment (dsherret) [inline cli/cache/mod.rs:284]: This function is only called for JSR specifiers that have module info stored in the metadata.

File: cli/tests/integration/publish_tests.rs [hunk 2]
```diff
@@ -47,6 +47,7 @@ itest!(javascript_missing_decl_file {
   envs: env_vars_for_registry(),
   exit_code: 0,
   temp_cwd: true,
+  http_server: true,
```
Comment (dsherret) [inline cli/tests/integration/publish_tests.rs:50]: This was missing and sometimes failing on the CI based on whether a test server was running or not

Comment (dsherret) [reply cli/tests/integration/publish_tests.rs:49]: I was so confused by the failing CI 😂

**Review by bartlomieju — APPROVED** [review]
LGTM

**Pull request merged**